Update Editor to 2023-10-17

Arnei opened the pull request:
The biggest (and most obvious) change is the graphical redesign, akin to #5318. Furthermore, this also makes the editor work with the changes to subtitles coming with Opencast 15.

This goes into develop as most agreed that the redesign is too big of a change to release it as a minor version.

Known major issues (may or may not be related to this release):
- opencast/opencast-editor#1154
- opencast/opencast-editor#1161

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

Arnei committed:
Update Editor to 2023-10-17

The biggest (and most obvious) change is the graphical redesign, akin to #5318.
Furthermore, this also makes the editor work with the changes to subtitles
coming with Opencast 15.

This goes into develop as most agreed that the redesign is too big of a change to release it as a minor version.

Known major issues (may or may not be related to this release):
- opencast/opencast-editor#1154
- opencast/opencast-editor#1161

KatrinIhler commented:
Out of interest: Will there be an update for OC 15 as well that contains the subtitle changes (but not the redesign), to make that work?

Arnei commented:
This PR contains the necessary changes to the editor frontend to work with the subtitle changes in OC 15. I'll admit the [release notes](https://github.com/opencast/opencast-editor/releases/tag/2023-10-17) don't really make that clear ("Change subtitle identification from flavor to id"  is the responsible PR).

KatrinIhler commented on etc/ui-config/mh_default_org/editor/editor-settings.toml:
```diff
@@ -71,18 +99,21 @@
-"captions/source+es" = "Español"
+## A list of languages for which new subtitles can be created
+# For each language, various tags can be specified
+# A list of officially recommended tags can be found at: TODO: link to opencast documentation for subtitle tags
```
Missing link :)

Arnei committed:
Add missing link to documentation

Adds a link to the subtitle documentation in opencast to
the editor configuration file.

KatrinIhler commented:
> This PR contains the necessary changes to the editor frontend to work with the subtitle changes in OC 15. I'll admit the [release notes](https://github.com/opencast/opencast-editor/releases/tag/2023-10-17) don't really make that clear ("Change subtitle identification from flavor to id" is the responsible PR).

Yes, sorry, got my releases mixed up, develop = OC 15. :sweat_smile:

KatrinIhler commented:
We've got this version running in our Frankenstein version at Bern and haven't experienced any major problems so far (apart from the known issues). Also this is aimed at develop, so we have a bit of time to fix stuff until OC 15 is released. -> Merging.
